src/test/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverageIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+    }
+
+    @Test
+    public void testIntervalListOverlappingUniqueBehavior() throws IOException {
+        final String expectedBaseName = "testIntervalListOverlappingUniqueBehavior"; //interval_summary files were not generated with gatk3 as they behave by interval merging
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+        output.delete();
+
+        ArgumentsBuilder cmd = new ArgumentsBuilder();
+        cmd.addReference(new File(hg38Reference));
+        cmd.addInput(new File(largeFileTestDir+"multiSampleSubsetted.bam"));
+        cmd.addArgument("min-base-quality","0");
+        cmd.addArgument("include-deletions");
+        cmd.addArgument("partition-type", "sample");
+        cmd.addArgument("omit-depth-output-at-each-base");
+        cmd.addOutput(output);
+
+        // We construct overlapping intervals to demonstrate the new interval merging behavior
+        for (int i = 1656275; i < 1656975; i+=10) {
+            cmd.addInterval(new SimpleInterval("chr1", i, i+100));
+        }
+        runCommandLine(cmd);
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+
+    @Test
+    // NOTE, the gene list file was not generated with GATK3 due to different gene list merging behavior
+    public void testGeneListDataPartallylCovered() throws IOException {
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String[] cmd = new String[]{ "-R",hg38Reference,
+                "-I",largeFileTestDir+"multiSampleSubsetted.bam",
+                "-L", "chr1:1656275-1677440", // this should only completely span two of the genes in the list
+                "--calculate-coverage-over-genes",getTestFile("refGene_CDK11B.refseq").getAbsolutePath(),
+                "--min-base-quality","0","-dels","-pt","sample","--output-format","CSV","--omit-depth-output-at-each-base","--omit-per-sample-statistics","--omit-genes-not-entirely-covered-by-traversal",
+                "-O",output.getAbsolutePath()};
+        runCommandLine(cmd);
+
+        try (final CSVReader actualParser = new CSVReader(new FileReader(output.getAbsolutePath()+".sample_gene_summary"))) {
+            actualParser.readNext(); // Skip header line
+            List<String[]> lines = actualParser.readAll();
+            Assert.assertEquals(lines.size(), 2);
+            // The only two genes completely encompassed by this interval span
+            Assert.assertEquals(lines.get(0)[0], "SLC35E2");
+            Assert.assertEquals(lines.get(1)[0], "SLC35E2");
+            Assert.assertNotEquals(lines.get(0), lines.get(1));
+        }
```

Comment:
You're not calling `compareOutputDirectories()` to fully check the output files here? Why not?